### PR TITLE
Yet more Clippy

### PR DESCRIPTION
**Pull request by nickdrozd**
There are a few changes packed into this one.

- Cut unnecessary negations (mostly shuffling if/else blocks)
- Use isize::min_value() instead of std::isize::MIN, etc
    - https://github.com/rust-lang/rust-clippy/issues/2380
- Rename some shadowing variables
- Cut some unnecessary references
- Use <struct>::default instead of Default::default
- Update inclusive range syntax
- Cut open import
- Self
- dyn trait

### CODE REVIEW

**Comment by nickdrozd** [comment]
I've been collecting these for a while, like a bird building a nest.

File: rust_src/src/editfns.rs [hunk 1]
```diff
@@ -458,15 +458,13 @@ pub fn byte_to_string(byte: EmacsInt) -> LispObject {
 /// Return the first character in STRING.
 #[lisp_fn]
 pub fn string_to_char(string: LispStringRef) -> EmacsInt {
```
Comment (shaleh) [inline rust_src/src/editfns.rs]: Good stuff

File: rust_src/src/filelock.rs [hunk 2]
```diff
@@ -25,7 +25,6 @@ use crate::{
     remacs_sys::{Qnil, Qstringp, Qt},
     threads::ThreadState,
 };
```
Comment (shaleh) [inline rust_src/src/filelock.rs]: Double win. No more asterisk and using Self.

File: rust_src/src/fns.rs [hunk 3]
```diff
@@ -591,10 +591,10 @@ pub fn copy_sequence(mut arg: LispObject) -> LispObject {
         let mut new = unsafe { make_uninit_bool_vector(nbits).force_bool_vector() };
         new.as_mut_slice().copy_from_slice(boolvec.as_slice());
         new.into()
```
Comment (shaleh) [inline rust_src/src/fns.rs]: Nice.

File: rust_src/src/fonts.rs [hunk 4]
```diff
@@ -117,7 +117,7 @@ impl LispFontObjectRef {
         unsafe { font_add_log(c_str.as_ptr(), self.into(), result.into()) }
     }
```
Comment (shaleh) [inline rust_src/src/fonts.rs]: `_frame` was used here because when `window-system` is not enabled the `frame` argument is unused and clippy rightly complains.

File: rust_src/src/frames.rs [hunk 5]
```diff
@@ -633,12 +633,12 @@ pub fn frame_focus(frame: LispFrameLiveOrSelected) -> LispObject {
 ///
 /// If there is no window system support, this function does nothing.
 #[lisp_fn(min = "1", name = "x-focus-frame", c_name = "x_focus_frame")]
```
Comment (shaleh) [inline rust_src/src/frames.rs]: Same explanation as above.

File: rust_src/src/multibyte.rs [hunk 6]
```diff
@@ -165,16 +166,17 @@ impl Codepoint {
     }
 
     // Equivalent to UNIBYTE_TO_CHAR
-    pub fn unibyte_to_char(self) -> Codepoint {
-        match self.is_ascii() {
-            true => self,
-            false => Codepoint::from_raw(self.0 as u8),
+    pub fn unibyte_to_char(self) -> Self {
+        if self.is_ascii() {
```
Comment (shaleh) [inline rust_src/src/multibyte.rs]: What is wrong with using `match` on the bool like before? Is it poor Rust style or was this a personal choice?

File: rust_src/src/windows.rs [hunk 7]
```diff
@@ -95,7 +95,7 @@ impl LispWindowRef {
 
         if mode_line_height >= 0 {
             mode_line_height
```
Comment (shaleh) [inline rust_src/src/windows.rs]: Why did you change from `!=` to `< 0`? The C code used:
```
#define CURRENT_MODE_LINE_HEIGHT(W)					\
  (W->mode_line_height >= 0						\
   ? W->mode_line_height						\
   : (W->mode_line_height						\
      = (MATRIX_MODE_LINE_HEIGHT (W->current_matrix)			\
	 ? MATRIX_MODE_LINE_HEIGHT (W->current_matrix)			\
	 : estimate_mode_line_height					\
	     (XFRAME (W->frame), CURRENT_MODE_LINE_FACE_ID (W)))))
```

Which means `MATRIX_MODE_LINE_HEIGHT` is only checked for being not 0. -1 and 1 are both true values as far as `if` is concerned. Only `0` is false.

Comment (nickdrozd) [reply rust_src/src/windows.rs]: I don't know, I must have misread something.

File: rust_src/src/multibyte.rs (hunk 6 above)
Comment (nickdrozd) [reply rust_src/src/multibyte.rs]: Clippy flags it. Not important of course, but matching on bools does seem like a strange style.

File: rust_src/src/fonts.rs (hunk 4 above)
Comment (nickdrozd) [reply rust_src/src/fonts.rs]: Clippy was complaining about that when I ran it locally, but I guess it just doesn't handle this situation very well.

File: rust_src/src/windows.rs (hunk 7 above)
Comment (shaleh) [reply rust_src/src/windows.rs]: I read it twice myself. The variable checked flips from the if to the else if which adds to the confusion.

**Comment by nickdrozd** [comment]
@shaleh I updated `rust-toolchain`, uncovering a few more flagged items.

File: rust_src/src/windows.rs (hunk 7 above)
Comment (nickdrozd) [reply rust_src/src/windows.rs]: This is fixed btw (that is, the change was removed entirely).

File: rust_src/src/marker.rs [hunk 8]
```diff
@@ -561,17 +561,20 @@ fn set_marker_internal_else(
...
             .map_or(false, |m| m.buffer() == Some(buf))
     {
-        bytepos = buf.charpos_to_bytepos(charpos);
+        buf.charpos_to_bytepos(charpos)
     } else {
-        let beg = buf.buffer_beg_byte(restricted);
-        let end = buf.buffer_end_byte(restricted);
-        bytepos = clip_to_bounds(beg, bytepos as EmacsInt, end);
-    }
+        clip_to_bounds(
```
Comment (shaleh) [inline rust_src/src/marker.rs]: A benefit to the temporary variables is they make it easier to inspect values in a debugger.
